R/pacta_sector_override_mapping.R

```diff
@@ -0,0 +1,499 @@
+#' Manually mantained PACTA Sector override mapping
+#' Use the `entity_proper_name` to match the company name in the FactSet
+#' database.
+#' For each company, the `pacta_sector` column contains the sector that
+#' should be used in the PACTA analysis.
+#' The `entity_id_md5` column contains the md5 hash of the entity_id, which
+#' is used to disambiguate the company in the FactSet database in the case of
+#' multiple companies with the same `entity_proper_name`.
```

Comment:
What about the case where a company name changes in the FactSet database? Will this then silently skip our override until we one day realize it's not doing what we thought it should?

Reply:
If the name changes in the DB, then the filter will cause that row to not be downloaded.

In the case of a single match, we'll see a change in the file contents (which is good, since it prompts us to investigate that mapping again)

In the case of a name change that causes multiple matches to appear, where previously there was only one (and the md5 is `NA`), then we'll get the `multiple_matches` block of code to activate, and get the warning message (along with the file changing)

In the case of previous multiple match going to a single match, then this bit of code:
```r
dplyr::mutate(
      keep_row = is.na(.data[["entity_id_md5"]]) |
        (.data[["fs_entity_id_md5"]] == .data[["entity_id_md5"]])
```
is doing the heavy lifting. If the "wrong" company changed names, then the one we want to keep still has the same entity ID, and the MD5 is still valid. If the remaining company with a name match is the "wrong one" then the row will get filtered out, and we'll get the "Company could not be matched" warnings. (as well as a file change)

In any case, since we're expecting this file to change infrequently, I'm expecting the downstream QA processes to flag any changes that need to get picked up in this repo (since comparison across time is out of scope for this)